### DISCUSSION TITLE
OAuth access token expiry

### DISCUSSION THREAD

**Discussion by j0sh** [pocketbase/pocketbase#3617] [Q&A]
Hi,

Related to https://github.com/pocketbase/pocketbase/discussions/2458 - if we store the OAuth access token ourselves, we don't know how long it is valid for.

The OAuth2.Token has an [Expiry](https://github.com/golang/oauth2/blob/4c91c17b32051e551fc27bdbdea4edc060b0aae5/token.go#L45-L50) field, perhaps this could be added to the Pocketbase [AuthUser struct](https://github.com/pocketbase/pocketbase/blob/94253f0dd5431e6a5676432537182bd2b0f2adae/tools/auth/auth.go#L12-L21) and the corresponding OAuth providers.

If that sounds sensible, I would be happy to add that in and send a PR. Right now I am only using the Github provider, so can easily test with that, but not sure about the others. I could also submit a PR updating just Github, and someone else can fill in the other providers as they are being updated / tested.

Another thought is to tie the Pocketbase JWT expiry to the access token expiry, but I imagine that might not be desirable for all cases. Maybe as a configurable option for OAuth collections? That would be a bigger / separate feature though.

For what it's worth. Github has a [fixed 8h token expiry](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-user-access-token-for-a-github-app#using-the-web-application-flow-to-generate-a-user-access-token) which makes this issue less urgent for my case. However, it would still be good to parameterize the expiry whenever possible, so we can properly manage refreshing as necessary.

**Comment by ganigeorgiev** [comment]
I've added it to the roadmap to consider adding the `Expiry` field in the meta response, but there are no plans for now to "sync" the PocketBase JWT `exp` claim with the OAuth2 token duration (if returned at all).
